cmd/state-offline-installer/install.go

```diff
@@ -89,62 +116,144 @@ func (r *runner) Run(params *Params) error {
+	configData, err := os.ReadFile(installerConfigPath)
+	if err != nil {
+		return errs.Wrap(err, "Failed to read config_file")
+	}
+
+	if err := json.Unmarshal([]byte(configData), &config); err != nil {
+		return errs.Wrap(err, "Failed to decode config_file")
+	}
+
+	installerDimensions := &dimensions.Values{
+		ProjectID: config.ProjectID,
+		CommitID:  config.CommitID,
+		Trigger:   p.StrP(target.TriggerCliOfflineInstaller.String()),
+	}
+	r.Event("start", installerDimensions)
+
 	/* Prompt for License */
 	licenseFileAssetPath := filepath.Join(assetsPath, licenseFileName)
 	{
 		b, err := fileutils.ReadFile(licenseFileAssetPath)
 		if err != nil {
-			return errs.Wrap(err, "Unable to open License file")
+			return r.handleFailure(err, "Unable to open License file", installerDimensions)
 		}
 
 		accepted, err := legalprompt.CustomLicense(string(b), r.out, r.prompt)
 		if err != nil {
-			return errs.Wrap(err, "Error with license acceptance")
+			return r.handleFailure(err, "Error with license acceptance", installerDimensions)
 		}
 		if !accepted {
-			return locale.NewInputError("License not accepted")
+			return r.handleAbort(
+				locale.NewInputError("License not accepted"),
+				"License failure",
+				installerDimensions,
+			)
 		}
 	}
 
 	/* Extract Artifacts */
 	artifactsPath := filepath.Join(tempDir, artifactsPathName)
 	if err := r.extractArtifacts(artifactsPath, assetsPath); err != nil {
-		return errs.Wrap(err, "Could not extract artifacts")
+		return r.handleFailure(err, "Could not extract artifacts", installerDimensions)
 	}
 
 	/* Install Artifacts */
 	offlineTarget := target.NewOfflineTarget(params.path, artifactsPath)
-	asrt, err := r.setupRuntime(artifactsPath, offlineTarget)
+	asrt, err := r.setupRuntime(artifactsPath, offlineTarget, logfile)
 	if err != nil {
-		return errs.Wrap(err, "Could not setup runtime")
+		return r.handleFailure(err, "Could not setup runtime", installerDimensions)
 	}
 
 	/* Manually Install License File */
 	{
 		err = fileutils.CopyFile(licenseFileAssetPath, filepath.Join(params.path, licenseFileName))
 		if err != nil {
-			return errs.Wrap(err, "Error copying license file")
+			return r.handleFailure(err, "Error copying license file", installerDimensions)
+		}
+	}
+
+	/* Manually Install config File */
+	{
+		err = fileutils.CopyFile(
+			installerConfigPath,
+			filepath.Join(params.path, installerConfigFileName),
+		)
+		if err != nil {
+			return r.handleFailure(err, "Error copying config file", installerDimensions)
+		}
+	}
+
+	var uninstallerSrc string
+	var uninstallerDest string
+
+	/* Manually Install uninstaller */
+	if rt.GOOS == "windows" {
+		/* shenanigans because windows won't let you delete an executable that's running */
+		installDir, err := filepath.Abs(params.path)
+		if err != nil {
+			return r.handleFailure(err, "Error determining absolute install directory", installerDimensions)
+		}
+		uninstallDir := filepath.Join(installDir, "uninstall-data")
```

Comment:
Nit: I think `params.path` and `installDir` are effectively the same right? Can we standardize on one or the other? It's a bit confusing at the moment.

Reply:
params.path is whatever the user entered (e.g `perl`, `../perl`, `\perl`) and installDir is the absolute path (e.g `C:\users\User\perl`, `/opt/perl`, `E:\perl`).  The former is OK in the context of the installation process but we need the latter in the context of the uninstall.  We'll need to use the former in the next phase of work (UX improvement) where we may need to echo it back in its raw form to the user.